rules-tests/TypeDeclaration/Rector/ClassMethod/ReturnTypeFromStrictFluentReturnRector/Fixture/non_final_class.php.inc

```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace Rector\Tests\TypeDeclaration\Rector\ClassMethod\ReturnTypeFromStrictFluentReturnRector\Fixture;
+
+class NonFinalClass
+{
+    public function test()
+    {
+        return $this;
+    }
+}
+
+?>
+-----
+<?php
+
+namespace Rector\Tests\TypeDeclaration\Rector\ClassMethod\ReturnTypeFromStrictFluentReturnRector\Fixture;
+
+class NonFinalClass
+{
+    public function test(): static
```

Comment:
@staabm when return `$this`, return `static` or return `self` is actually same, see

- https://3v4l.org/U6gfm#v7.0.33 on PHP 7.0
- https://3v4l.org/B61Io#v8.0.29 on PHP 8.0 (support return static)

so the fallback to return `self` is fine on php 7.0

Reply:
I added test for php 7.x add return `self` instead of `static` https://github.com/rectorphp/rector-src/pull/4890/commits/5e5ba5e680d10103d0e993fd80c4035c4af7b176